nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/main/java/org/apache/nifi/processors/hadoop/GetHDFSFileInfo.java

```diff
@@ -178,6 +183,15 @@
             .defaultValue(GROUP_ALL.getValue())
             .build();
 
+    public static final PropertyDescriptor BATCH_SIZE = new PropertyDescriptor.Builder()
+            .displayName("Batch Size")
+            .name("gethdfsfileinfo-batch-size")
+            .description("Number of records to put into an output flowfile when 'Destination' is set to 'Content'"
+                + " and 'Group Results' is set to 'None'")
+            .required(false)
+            .addValidator(StandardValidators.createAllowEmptyValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR))
```

Comment:
Can you explain the logic here @tpalfy ? I don't understand why we would ever allow an empty string for this property. It is marked as optional (required=false). So either it should be specified as an integer or not specified. It looks like there is quite a bit of code in this PR (creation of this validator, and several places where it checks if the value is empty) that can be avoided.